packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -225,28 +225,32 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         });
     }
 
-    public async read(blobid: string): Promise<string> {
-        return this.readWithEncodingOutput(blobid, "base64");
+    public async read(blobId: string): Promise<string> {
+        return this.readWithEncodingOutput(blobId, "base64");
     }
 
     /**
      * {@inheritDoc @fluidframework/driver-definitions#IDocumentStorageService.readString}
      */
-    public async readString(blobid: string): Promise<string> {
-        return this.readWithEncodingOutput(blobid, "string");
+    public async readString(blobId: string): Promise<string> {
+        return this.readWithEncodingOutput(blobId, "string");
     }
 
-    private async readWithEncodingOutput(blobid: string, outputFormat: string) {
-        let blob = this.blobCache.get(blobid);
+    private async readWithEncodingOutput(blobId: string, outputFormat: string): Promise<string> {
+        // const blob = this.blobCache.get(blobId);
+        let blob: IBlob | undefined;
+        assert(!blob || blob.encoding === "base64" || blob.encoding === undefined, "wrong blob encoding format");
+        let buffer;
         // Reset the timer on attempted cache read
         this.scheduleClearBlobsCache();
         if (blob === undefined) {
-            this.checkSnapshotUrl();
+            this.checkAttachmentGETUrl();
```

Comment:
Any chance we can start removing these kinds of APIs (where possible) in favor of getters that do same function - throw if uri is not present?